build.sh

```diff
@@ -20,3 +20,7 @@ buildImages
 if is_publish_images; then
     publishImages
 fi
+
+if is_publish_images_to_quay; then
```

Comment:
why are we not reusing publishImages ?

Reply:
- as I understand, `publishImages` is used for "manual" build - it asks the password from stdin and uses an image registry that is already configured in the system
- `publishImagesToQuay` is for calling from automation scripts - it doesn't care about login and hardcoded to quay for Che-Theia images

`publishImagesToQuay` "inspired by" https://github.com/eclipse/che-theia/blob/9ac5031d760b304f970a9a8bd3ec340f4f57dd2a/cico_common.sh#L53

I'm not sure if someone actually uses [`publishImages`](https://github.com/eclipse/che-theia/blob/124b2b21d6a3f3d2eb5bdd7dba8079ffeb4bfa25/build.include#L108-L121). Maybe we can remove it.

Reply:
we no longer publish to docker.io (due to quota/etc limits)
And i'm not sure it's good to have a docker login in build.include anyway

so let's make publish do only publishing (and if we're not logged in, it will throw an error but it's fine as build script should not manage "authentication on docker registries"

Reply:
Agree. Updated.